### PR TITLE
Run dotnet core buildpack test with published app versions for different stacks

### PR DESCRIPTION
**Pull request by mariash**
### Are you submitting this PR against the [develop branch](https://github.com/cloudfoundry/cf-acceptance-tests/tree/develop)?

Yes

### What is this change about?

dotnet core buildpack will reach NuGet registry for unpublished apps to get dotnet core library. That makes this test fail in internetless environment. Switch to using published app (e.g. vendor everything). Published apps need to be compiled for the runtime they will be running on, so there are 2 published versions included for cflinuxfs3 and cflinuxfs4. 

Makefile is provided to help rebuild those versions in case of code changes.

### Please provide contextual information.

Slack thread https://vmware.slack.com/archives/CUR2H2KFE/p1663189449408889

### What version of cf-deployment have you run this cf-acceptance-test change against?

e8130a2c616c4c27e215030e571dc441409f9f5b

### Please check all that apply for this PR:

- [ ] introduces a new test --- Are you sure everyone should be running this test?
- [X] changes an existing test
- [ ] requires an update to a CATs integration-config

### Did you update the README as appropriate for this change?

- [ ] YES
- [X] N/A

### How many more (or fewer) seconds of runtime will this change introduce to CATs?

It actually makes it faster because it skips compilation and fetching dependencies during staging.

### What is the level of urgency for publishing this change?

- [ ] **Urgent** - unblocks current or future work
- [X] **Slightly Less than Urgent**

### CODE REVIEW

File: detect/buildpacks.go [hunk 1]
```diff
@@ -116,13 +117,19 @@ var _ = DetectDescribe("Buildpacks", func() {
...
-				return helpers.CurlAppRoot(Config, appName)
-			}).Should(ContainSubstring("Hello from dotnet-core"))
-		})
+		for _, stack := range Config.GetStacks() {
+			It(fmt.Sprintf("[%s] makes the app reachable via its bound route", stack), func() {
+				assetPath, ok := assets.NewAssets().DotnetCore[stack]
+				if !ok {
+					Fail(fmt.Sprintf("dotnet-core app is missing asset for %s stack", stack))
+				}
+				Expect(cf.Cf("push", appName, "-m", DEFAULT_MEMORY_LIMIT, "-p", assetPath).Wait(Config.DetectTimeoutDuration())).To(Exit(0))
```
Comment (ctlong) [inline detect/buildpacks.go:126]: I think it makes sense to add a `-s` flag to specify the specific stack to use, right?

File: detect/buildpacks.go [hunk 2]
```diff
@@ -116,13 +117,19 @@ var _ = DetectDescribe("Buildpacks", func() {
...
 	Describe("dotnet-core", func() {
-		It("makes the app reachable via its bound route", func() {
-			Expect(cf.Cf("push", appName, "-m", DEFAULT_MEMORY_LIMIT, "-p", assets.NewAssets().DotnetCore).Wait(Config.DetectTimeoutDuration())).To(Exit(0))
-
-			Eventually(func() string {
-				return helpers.CurlAppRoot(Config, appName)
-			}).Should(ContainSubstring("Hello from dotnet-core"))
-		})
+		for _, stack := range Config.GetStacks() {
+			It(fmt.Sprintf("[%s] makes the app reachable via its bound route", stack), func() {
```
Comment (ctlong) [inline detect/buildpacks.go:121]: `stack` needs to be assigned to a local copy inside of the for loop or else "the It closure will capture the mutating loop variable and all the specs will run against the last element in the loop." (re: [Dynamically Generating Specs](https://onsi.github.io/ginkgo/#dynamically-generating-specs))

Comment (ctlong) [inline detect/buildpacks.go:121]: Can you please either put `stack` in a Context title or rephrase the It title to something more like `makes the app reachable via its bound route in <stack>`.

File: detect/buildpacks.go [hunk 3]
```diff
@@ -116,13 +117,19 @@ var _ = DetectDescribe("Buildpacks", func() {
...
+				assetPath, ok := assets.NewAssets().DotnetCore[stack]
+				if !ok {
```
Comment (ctlong) [inline detect/buildpacks.go:123]: Please change this to an expectation, i.e. `Expect(ok).To(BeTrue())`. You can still add your current failure message to that [as an annotation](https://onsi.github.io/gomega/#annotating-assertions).

**Review by ctlong — CHANGES_REQUESTED** [review]
In general, I'm a little wary of the dynamic generation approach you've taken, as it's different to the static approach generally taken in CATs, but if it works I'm open to it :).

The change to the dotnet-core app looks good to me. I just have a few comments for the detect suite changes that need to be addressed.

**Comment by mariash** [comment]
I think in general cats should provide a way to test against different stacks. The configuration for CATs allows to set a list of stacks to test against, but in reality we only test against one. But this would be a bigger change and would affect many tests. As of now I only changed the dotnet test because we have different pre-compiled assets for dotnet app based on a stack.

**Comment by mariash** [comment]
@ctlong it looks like I can no longer push to the branch, I opened a new PR from my fork. https://github.com/cloudfoundry/cf-acceptance-tests/pull/610

**Pull request closed without merging**